flink-ml-benchmark/src/main/java/org/apache/flink/ml/benchmark/datagenerator/common/ColsWithLabelGenerator.java

```diff
@@ -0,0 +1,123 @@
+
+/** A DataGenerator which creates a table of column features with label. */
+public class ColsWithLabelGenerator
+        implements InputDataGenerator<ColsWithLabelGenerator>,
+        HasVectorDim<ColsWithLabelGenerator> {
+
+    private final Map<Param<?>, Object> paramMap = new HashMap<>();
+
+    public ColsWithLabelGenerator() {
+        ParamUtils.initializeMapWithDefaultValues(paramMap, this);
+    }
+
+    @Override
+    public Table[] getData(StreamTableEnvironment tEnv) {
+
+        StreamExecutionEnvironment env = TableUtils.getExecutionEnvironment(tEnv);
+        TypeInformation<?>[] types;
+        String[] colNames;
+        if (getColNames().length == 2) {
+            colNames = new String[getColNames()[0].length + 1];
+
+            types = new TypeInformation<?>[getColNames()[0].length + 1];
+            for (int i = 0; i < getColNames()[0].length; ++i) {
+                colNames[i] = getColNames()[0][i];
+                types[i] = Types.DOUBLE;
+            }
+            colNames[colNames.length - 1] = getColNames()[1][0];
+            types[colNames.length - 1] = Types.DOUBLE;
+        } else {
+            colNames = getColNames()[0];
+            types = new TypeInformation<?>[getColNames()[0].length];
+            for (int i = 0; i < colNames.length; ++i) {
+                types[i] = Types.DOUBLE;
+            }
+        }
+        DataStream<Row> dataStream =
+                env.fromParallelCollection(
+                                new NumberSequenceIterator(1L, getNumValues()),
+                                BasicTypeInfo.LONG_TYPE_INFO)
+                        .map(
+                                new ColsGenerator(getSeed(), colNames),
+                                new RowTypeInfo(types, colNames));
+
+        Table dataTable = tEnv.fromDataStream(dataStream);
+
+        return new Table[] {dataTable};
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    private static class ColsGenerator extends RichMapFunction<Long, Row> {
+        private final long initSeed;
+        private final String[] colNames;
+        private Random random;
+
+        private ColsGenerator(long initSeed, String[] colNames) {
+            this.initSeed = initSeed;
+            this.colNames = colNames;
+        }
+
+        @Override
+        public void open(Configuration parameters) throws Exception {
+            super.open(parameters);
+            int index = getRuntimeContext().getIndexOfThisSubtask();
+            random = new Random(Tuple2.of(initSeed, index).hashCode());
+        }
+
+        @Override
+        public Row map(Long ignored) {
+            Row ret = new Row(colNames.length);
+            for (int i = 0; i < colNames.length - 1; ++i) {
+                ret.setField(i, random.nextDouble());
+            }
+            ret.setField(colNames.length - 1, random.nextDouble() > 0.5 ? 1.0 : 0.0);
+            return ret;
+        }
+    }
+}
```

Comment:
nit: Let's follow the common practice to add a blank line at the end of each file.